packages/cli/src/util/rollback/status.ts

```diff
@@ -0,0 +1,255 @@
+      if (jobStatus !== 'pending' && jobStatus !== 'in-progress') {
+        output.log(`Unknown rollback status "${jobStatus}"`);
+        return 1;
+      }
+
+      // check if we have been running for too long
+      if (requestedAt < recentThreshold || Date.now() >= rollbackTimeout) {
+        output.log(
+          `The rollback exceeded its deadline - rerun ${chalk.bold(
+            `vercel rollback ${toDeploymentId}`
+          )} to try again`
+        );
+        return 1;
+      }
+
+      // if we've done our first poll and not rolling back, then print the
+      // requested at date/time
+      if (counter++ === 0 && !deployment) {
+        spinnerMessage += ` requested at ${formatDate(requestedAt)}`;
+      }
+      output.spinner(`${spinnerMessage}…`);
+
+      await sleep(250);
+    }
+  } finally {
+    output.stopSpinner();
+  }
+}
+
+async function renderJobFailed({
+  client,
+  contextName,
+  deployment,
+  project,
+  toDeploymentId,
+}: {
+  client: Client;
+  contextName: string;
+  deployment?: Deployment;
+  project: Project;
+  toDeploymentId: string;
+}) {
+  const { output } = client;
+
+  try {
+    const name = (
+      deployment ||
+      (await getDeploymentInfo(client, contextName, toDeploymentId))
+    )?.url;
+    output.error(
+      `Failed to remap all aliases to the requested deployment ${name} (${toDeploymentId})`
+    );
+  } catch (e) {
+    output.error(
+      `Failed to remap all aliases to the requested deployment ${toDeploymentId}`
+    );
+  }
+
+  // aliases are paginated, so continuously loop until all of them have been
+  // fetched
+  let nextTimestamp;
+  for (;;) {
+    let url = `/v9/projects/${project.id}/rollback/aliases?failedOnly=true&limit=20`;
+    if (nextTimestamp) {
+      url += `&until=${nextTimestamp}`;
+    }
+
+    const { aliases, pagination } = await client.fetch<RollbackAliasesResponse>(
+      url
+    );
+
+    for (const { alias, status } of aliases) {
+      output.log(
+        `  ${renderAliasStatus(status).padEnd(11)}  ${alias.alias} (${
+          alias.deploymentId
+        })`
+      );
+    }
+
+    if (pagination?.next) {
+      nextTimestamp = pagination.next;
+    } else {
+      break;
+    }
+  }
+
+  return 1;
+}
+
+async function renderJobSucceeded({
+  client,
+  contextName,
+  performingRollback,
+  project,
+  requestedAt,
+  toDeploymentId,
+}: {
+  client: Client;
+  contextName: string;
+  performingRollback: boolean;
+  project: Project;
+  requestedAt: number;
+  toDeploymentId: string;
+}) {
+  const { output } = client;
+
+  let deploymentInfo = '';
+  try {
+    const deployment = await getDeploymentInfo(
+      client,
+      contextName,
+      toDeploymentId
+    );
+    deploymentInfo = `${chalk.bold(deployment.url)} (${toDeploymentId})`;
+  } catch (err: unknown) {
+    if (err instanceof Error) {
+      output.debug(
+        `Failed to get deployment url for ${toDeploymentId}: ${err.toString()}`
+      );
+    }
```

Comment:
**suggestion (non-blocking):** remove `instanceof` check here as well?

Reply:
Fixed. I do this to make TypeScript happy, but I can make TypeScript happy without the instanceof.